src/query/stream/streams.cpp

```diff
@@ -232,14 +236,23 @@ Streams::StreamsMap::iterator Streams::CreateConsumer(StreamsMap &map, const std
         spdlog::trace("Commit transaction in stream '{}'", stream_name);
         interpreter->CommitTransaction();
         result.rows.clear();
-      };
-
-  auto insert_result = map.try_emplace(
-      stream_name, StreamData<TStream>{std::move(stream_info.common_info.transformation_name), std::move(owner),
-                                       std::make_unique<SynchronizedStreamSource<TStream>>(
-                                           stream_name, std::move(stream_info), std::move(consumer_function))});
-  MG_ASSERT(insert_result.second, "Unexpected error during storing consumer '{}'", stream_name);
-  return insert_result.first;
+      } catch (const query::TransactionSerializationException &e) {
+        const bool has_retry_limit_reached = (i + 1) == total_retries;
```

Comment:
```
while (i < total_retries) {
 try {
  ...
 } catch (...) {
   ++i;
   const bool has_retry_limit_reached = i == total_retries;
   ...
 }
```
To avoid increasing `i` twice :smile: